Bottom: 762d4cc288489661211fe2ead1316ede454959d3
Top:    422984482292d896b9801f977c7465d76e4e7ebd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-26 22:36:35 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index cb40e4d..86235d2 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -307,6 +307,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
